snippets/product-variant-options.liquid

```diff
@@ -45,10 +47,47 @@
     endfor
   -%}
 
-  {%- if block.settings.picker_type == 'button' -%}
+  {%- liquid
+    if value.swatch.image
+      assign image_url = value.swatch.image | image_url: width: 50
+      assign swatch_value = 'url(' | append: image_url | append: ')'
+    elsif value.swatch.color
+      assign swatch_value = 'rgb(' | append: value.swatch.color.rgb | append: ')'
+    else
+      assign swatch_value = nil
+    endif
+  -%}
```

Comment:
I wonder if that should just be part of the previous `{%- liquid ...` content 🤔 

Though maybe it's nice to differentiate the two goals.

Reply:
Updated!